From 56f518855f76eddd9632fd54ef56c0f23b2d1c87 Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:50 -0400
Subject: [virt] kvm/ppc: book3s - Move KVM_REG_PPC_WORT to an unused register number

Message-id: <1410765214-16377-35-git-send-email-dgibson@redhat.com>
Patchwork-id: 94731
O-Subject: [PATCH 34/78] KVM: PPC: Book3S: Move KVM_REG_PPC_WORT to an unused register number
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Signed-off-by: David Gibson <dgibson@redhat.com>

commit e1d8a96daf641aea11c25268eab678a76215541a
Author: Paul Mackerras <paulus@samba.org>
Date:   Mon May 26 19:48:35 2014 +1000

    KVM: PPC: Book3S: Move KVM_REG_PPC_WORT to an unused register number

    Commit b005255e12a3 ("KVM: PPC: Book3S HV: Context-switch new POWER8
    SPRs") added a definition of KVM_REG_PPC_WORT with the same register
    number as the existing KVM_REG_PPC_VRSAVE (though in fact the
    definitions are not identical because of the different register sizes.)

    For clarity, this moves KVM_REG_PPC_WORT to the next unused number,
    and also adds it to api.txt.

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/virtual/kvm/api.txt b/Documentation/virtual/kvm/api.txt
index cd9f2cc..455e552 100644
--- a/Documentation/virtual/kvm/api.txt
+++ b/Documentation/virtual/kvm/api.txt
@@ -1859,6 +1859,7 @@ registers, find a list below:
   PPC   | KVM_REG_PPC_PPR | 64
   PPC   | KVM_REG_PPC_ARCH_COMPAT 32
   PPC   | KVM_REG_PPC_DABRX     | 32
+  PPC   | KVM_REG_PPC_WORT      | 64
   PPC   | KVM_REG_PPC_TM_GPR0 | 64
           ...
   PPC   | KVM_REG_PPC_TM_GPR31 | 64
diff --git a/arch/powerpc/include/uapi/asm/kvm.h b/arch/powerpc/include/uapi/asm/kvm.h
index e45bdbf..6463215 100644
--- a/arch/powerpc/include/uapi/asm/kvm.h
+++ b/arch/powerpc/include/uapi/asm/kvm.h
@@ -531,7 +531,6 @@ struct kvm_get_htab_header {
 #define KVM_REG_PPC_TCSCR (KVM_REG_PPC | KVM_REG_SIZE_U64 | 0xb1)
 #define KVM_REG_PPC_PID  (KVM_REG_PPC | KVM_REG_SIZE_U64 | 0xb2)
 #define KVM_REG_PPC_ACOP (KVM_REG_PPC | KVM_REG_SIZE_U64 | 0xb3)
-#define KVM_REG_PPC_WORT (KVM_REG_PPC | KVM_REG_SIZE_U64 | 0xb4)
 
 #define KVM_REG_PPC_VRSAVE (KVM_REG_PPC | KVM_REG_SIZE_U32 | 0xb4)
 #define KVM_REG_PPC_LPCR (KVM_REG_PPC | KVM_REG_SIZE_U32 | 0xb5)
@@ -541,6 +540,7 @@ struct kvm_get_htab_header {
 #define KVM_REG_PPC_ARCH_COMPAT (KVM_REG_PPC | KVM_REG_SIZE_U32 | 0xb7)
 
 #define KVM_REG_PPC_DABRX (KVM_REG_PPC | KVM_REG_SIZE_U32 | 0xb8)
+#define KVM_REG_PPC_WORT (KVM_REG_PPC | KVM_REG_SIZE_U64 | 0xb9)
 
 /* Transactional Memory checkpointed state:
  * This is all GPRs, all VSX regs and a subset of SPRs
-- 
1.7.1